src/App.tsx

```diff
@@ -101,13 +81,13 @@ export const App: React.FC = () => {
               'is-8-desktop',
               'Sidebar',
               {
-                'Sidebar--open': selectedPost,
+                'Sidebar--open': selectedPost.post,
               },
             )}
           >
             <div className="tile is-child box is-success ">
-              {selectedPost && (
-                <PostDetails post={selectedPost} />
+              {selectedPost.post && selectedPost.post.id !== null && (
```

Comment:
Use destructuring for the selectedPost object and use logical operators 
```suggestion
              {post && !!post.id && (
```